### PR TITLE
feat: take tick range from URL

### PR DESCRIPTION
**Pull request by dmihal**
Lets an outside application pre-set the price range, for example:

`/add/ETH/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/3000?minPrice=100&maxPrice=500`

### CODE REVIEW

**Comment by dmihal** [comment]
Working test link:
https://interface-git-fork-dmihal-tick-range-uniswap.vercel.app/#/add/ETH/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/3000?chain=mainnet&minPrice=2000&maxPrice=3000

**Comment by JFrankfurt** [comment]
I love this. @tinaszheng what do you think of the behavior here? @dmihal has the effect removing the param instead of maintaining state sync w/ the ui like we do for fee level/asset/network. My gut says we should sync as the user changes so the url is shareable into a specific range.

File: src/pages/AddLiquidity/index.tsx [hunk 1]
```diff
@@ -150,6 +150,31 @@ export default function AddLiquidity({
 
   useEffect(() => setShowCapitalEfficiencyWarning(false), [baseCurrency, quoteCurrency, feeAmount])
 
+  useEffect(() => {
+    if (history.location.search) {
+      const search = new URLSearchParams(history.location.search)
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:155]: we have a `useParsedQueryString()` hook

File: src/pages/AddLiquidity/index.tsx [hunk 2]
```diff
@@ -150,6 +150,31 @@ export default function AddLiquidity({
...
+
+      const maxPrice = search.get('maxPrice')
+      if (maxPrice) {
+        updated = true
+        onRightRangeInput(maxPrice)
+        search.delete('maxPrice')
+      }
+
+      if (updated) {
+        history.replace({ search: search.toString() })
```
Comment (moodysalem) [inline src/pages/AddLiquidity/index.tsx:173]: it seems like this would cause an infinite loop?

**Comment by tinaszheng** [comment]
ooo thanks so much for your contribution @dmihal !! looks great but i do agree that we should probably persist the query params and also sync the URL with the app ranges. you can take a look at https://github.com/Uniswap/interface/blob/main/src/components/Header/NetworkSelector.tsx#L284 for some inspiration for how we sync network params with the app (although that one is probably more complicated)

File: src/pages/AddLiquidity/index.tsx (hunk 1 above)
Comment (dmihal) [reply src/pages/AddLiquidity/index.tsx]: Cool, I've switched to using it

File: src/pages/AddLiquidity/index.tsx (hunk 2 above)
Comment (dmihal) [reply src/pages/AddLiquidity/index.tsx]: Updated, query params now stay in the URL

**Comment by dmihal** [comment]
Hey everyone, thanks for the review!

I've updated the PR to persist the URL param, so users can easily have "sharable" URLs, similar to the network param shared by @tinaszheng

I've also added  useParsedQueryString at @moodysalem's suggestion.

File: src/pages/AddLiquidity/index.tsx [hunk 3]
```diff
@@ -150,6 +153,16 @@ export default function AddLiquidity({
 
   useEffect(() => setShowCapitalEfficiencyWarning(false), [baseCurrency, quoteCurrency, feeAmount])
 
+  useEffect(() => {
+    if (typeof parsedQs.minPrice === 'string' && parsedQs.minPrice !== leftRangeTypedValue) {
+      onLeftRangeInput(parsedQs.minPrice)
+    }
+
+    if (typeof parsedQs.maxPrice === 'string' && parsedQs.minPrice !== rightRangeTypedValue) {
```
Comment (tinaszheng) [inline src/pages/AddLiquidity/index.tsx:161]: i believe you want `parsedQs.maxPrice` here?

**Review by tinaszheng — CHANGES_REQUESTED** [review]
just 2 requested changes - otherwise looks good so far!

File: src/pages/AddLiquidity/index.tsx (hunk 3 above)
Comment (dmihal) [reply src/pages/AddLiquidity/index.tsx:161]: Good catch!

**Review by JFrankfurt — APPROVED** [review]
lgtm -- I noticed though that some ticks end up with a bunch of trailing 0's, while others do not. (i.e., `40.100000` and `60.2`)

**Review by tinaszheng — APPROVED** [review]

**Comment by JFrankfurt** [comment]
thanks @dmihal !

**Comment by dmihal** [comment]
Thanks for merging this @JFrankfurt & @tinaszheng!

Great timing too, we just had a team call today, and hopefully launching the staker soon!

**Pull request merged**